src/sev-snp/sev_support.cc

```diff
@@ -342,7 +364,7 @@ int sev_request_key(struct sev_key_options *options,
   }
 
   errno = 0;
-  rc = ioctl(fd, SNP_GET_DERIVED_KEY, &guest_req);
+  ioctl_with_retry(fd, SNP_GET_DERIVED_KEY, &rc, &guest_req);
```

Comment:
you macro doesn't return -1 on error and directly returns `errno`; this might be a problem because on the next line we compare with -1 to detect errors

Reply:
Will fix. Thanks!